### DISCUSSION TITLE
Converting semi transparent colour to opaque

### DISCUSSION THREAD

**Discussion by SitKneesweak** [ImageMagick/ImageMagick#6411] [Help]
### ImageMagick version

7.1.1

### Operating system, version and so on

Windows 10

### Description

I have a PNG file which has blue in varying alphas with a transparent background.

How can I keep the transparent background but turn all the blue colours to 100% opaque?

<img width="322" alt="blue" src="https://github.com/ImageMagick/ImageMagick/assets/122905247/81206a98-bf8a-4dce-ab72-c5a7b99529b1">

**Comment by fmw42** [comment]
Your alpha channel is totally opaque!  There is no transparency on this file.

**Comment by SitKneesweak** [comment]
It could be because I opened it in photo viewer, here it is in the package it was drawn in, Aseprite:
<img width="416" alt="blu2" src="https://github.com/ImageMagick/ImageMagick/assets/122905247/25bc6c63-f879-4c3a-9dd6-3c40427a107f">

**Comment by snibgo** [comment]
Please show the input file. The actual file, not a screenshot of some program displaying the file.

**Comment by SitKneesweak** [comment]
![dustLogical](https://github.com/ImageMagick/ImageMagick/assets/122905247/bd4e343c-b1f5-4f50-b55f-596ee67a854d)

**Comment by snibgo** [comment]
```
magick in.png -channel A -threshold 0 +channel out.png
```
This leaves fully-transparent pixels fully transparent. Partly transparent pixels are changed to fully opaque.

**Comment by SitKneesweak** [comment]
Beautiful, shall test in a few hours!

**Reply by SitKneesweak** [reply]
Amazing, it works, thanks very much
